presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -109,7 +109,9 @@
 
     private int[] values;
     private ResultsProcessor resultsProcessor = new ResultsProcessor();
-    private int averageResultSize = SIZE_OF_DOUBLE;
+    private int averageResultSize = SIZE_OF_INT;
+    private boolean[] inDictionaryFlags;
+    private boolean anyRowGroupDictionaries;
```

Comment:
- this variable name is confusing; I think it is set if current row group has a dictionary; if so, let's rename to rowGroupDictionaryFound or rowGroupDictionaryPresent or hasRowGroupDictionary
- the lifecycle of this variable is also confusing; I'd expect it to be reset in startStripe and openRowGroup methods, but it is only set to true in openRowGroup. The reset appears to be happening in mustExtractValues. 

It might be helpful to rename mustExtractValues to mustExtractValuesBeforeNextRowGroup and change its parameter name to `newStripe`. This would make it clear that the method is called before each new row group and the parameter further clarifies whether the new row groups appears in a new stripe or not.

`boolean mustExtractValuesBeforeNextRowGroup(boolean newStripe)`